src/algorithms/audioproblems/discontinuitydetector.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_DISCONTINUITYDETECTOR_H
+#define ESSENTIA_DISCONTINUITYDETECTOR_H
+
+#include "algorithm.h"
+#include "algorithmfactory.h"
+#include "essentiamath.h"
+
+namespace essentia {
+namespace standard {
+
+class DiscontinuityDetector : public Algorithm {
+
+  private:
+    Input<std::vector<Real>> _frame;
+    Output<std::vector<Real>> _discontinuityLocations;
+    Output<std::vector<Real>> _discontinuityAmplitues;
+
+    int _order;
+    int _hopSize;
+    int _kernelSize;
+    float _detectionThld;
+    float _energyThld;
+    int _subFrameSize;
+    int _frameSize;
+
+    Algorithm* _medianFilter;
+    Algorithm* _LPC;
+    Algorithm* _windowing;
+
+    // normalize to the max(abs(array))
+    template <typename T> void absNormalize(std::vector<T>& array) {
+      if (array.empty()) return;
+      std::vector<T> absArray = array;
+      rectify(absArray);
+      T maxElement = *std::max_element(absArray.begin(), absArray.end());
+
+      if (maxElement != (T) 0.0) {
+        for (uint i=0; i<array.size(); i++) {
+          array[i] /= maxElement;
+        }
+      }
+    }
+
+  public:
+    DiscontinuityDetector() {
+        declareInput(_frame, "frame", "the input frame (must be non-empty)");
+        declareOutput(_discontinuityLocations, "discontinuityLocations", "the index of the detected discontinuities (if any)");
+        declareOutput(_discontinuityAmplitues, "discontinuityAmplitudes", "the peak values of the prediction error for the discontinuities (if any)");
+    
+      _medianFilter = AlgorithmFactory::create("MedianFilter");
+      _LPC = AlgorithmFactory::create("LPC");
+      _windowing = AlgorithmFactory::create("Windowing");
+    }
+
+    ~DiscontinuityDetector() {
+      if (_medianFilter) delete _medianFilter;
+      if (_LPC) delete _LPC;
+      if (_windowing) delete _windowing;
+    }
+
+    void declareParameters() {
+        declareParameter("order", "scalar giving the number of LPCs to use", "[1,inf)", 3);
+        declareParameter("frameSize", "frame size used for the analysis. Only discontinuities outside the overlapped areas are returned", "[0,inf)", 512);
```

Comment:
How frameSize relates to the input frame? From "frame size used for the analysis" it is not clear that this is actually the size of the input frame. Mention that frameSize is expected to be matching the size of the input frame signal, but that it is optional ([see example](https://github.com/MTG/essentia/blob/master/src/algorithms/standard/spectrum.h#L54))